test/extensions/filters/http/buffer/buffer_filter_fuzz_test.cc

```diff
@@ -0,0 +1,41 @@
+#include <chrono>
+#include <memory>
+
+#include "envoy/config/filter/http/buffer/v2/buffer.pb.h"
+
+#include "extensions/filters/http/buffer/buffer_filter.h"
+#include "extensions/filters/http/well_known_names.h"
+
+#include "test/extensions/filters/http/buffer/buffer_filter_fuzz.pb.h"
+#include "test/extensions/filters/http/common/fuzz/decoder_filter_fuzz.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/mocks/buffer/mocks.h"
+#include "test/mocks/http/mocks.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace BufferFilter {
+
+class BufferFilterFuzz : public DecoderFuzzFilter {
+public:
+  BufferFilterFuzz(envoy::config::filter::http::buffer::v2::Buffer proto_config)
+      : config_(std::make_shared<BufferFilterConfig>(proto_config)) {
+    // Initialize filter and setup callbacks.
+    filter_ = std::make_unique<BufferFilter>(config_);
+    filter_->setDecoderFilterCallbacks(callbacks_);
+  }
+
+  BufferFilterConfigSharedPtr config_;
+};
+
+DEFINE_PROTO_FUZZER(
+    const test::extensions::filters::http::buffer::BufferFilterFuzzTestCase& input) {
+  BufferFilterFuzz filter(input.config());
```

Comment:
Another way is:

class DecoderFuzzer {
public:
     DecoderFuzz(DecodeFilter* filer) : filter_(filter) {
        filter_->setDcoderFilterCallbacks(calback_);
   }

   void fuzz() {
       filter_->...
   }

   DocoderFilter* filter_{};
   callback_;
};

Each custom filter not need to derive from it. 

DEFINE_PROTO_FUZZER()  {
        filter = create custom filter;
        fuzzer = DoecoerFuzzer(&filter);
        fuzzer.fuzz(input.data());
}

Reply:
^ Ooh. I like this a lot more. It leaves the filter fuzz writer with less responsibility to look at the internal of the `DecoderFuzzer` class. I'll update the POC.